patches/tModLoader/Terraria/NPC.cs.patch

```diff
@@ -51,8 +51,84 @@
++	/// </summary>
+-	private static float gravity = 0.3f;
++	//private static float gravity = 0.3f;
++	public float gravity {
++		get => internalGravityBase * Gravity.Value;
++		private set {
++			Gravity = MultipliableFloat.One;
++			internalGravityBase = value;
++		}
++	}
++
++	/// <summary>
++	/// Used to keep vanilla and modded gravity effects working neatly
++	/// </summary>
++	private float internalMaxFallSpeedBase = 10f; //TML: #AccessibleNPCGravity, added
++
++	/// <summary>
++	/// Modify this value in order to change the NPCs active maxFallSpeed, this can be done in AI as gravity values are reset slightly beforehand
++	/// </summary>
++	public MultipliableFloat MaxFallSpeed = MultipliableFloat.One; //TML: #AccessibleNPCGravity, added
++
++	/// <summary>
++	/// The current actual fall speed cap in velocity applied every frame. Returns the combination of tMod internal values and <see cref="MaxFallSpeed"/> that will actually be used to modify velocity. Do not use unless there is a specific reason why as this may causes issues with other mods, use <see cref="MaxFallSpeed"/> instead.
++	/// </summary>
++	public float maxFallSpeed { //TML: #AccessibleNPCGravity, added
++		get => internalMaxFallSpeedBase * MaxFallSpeed.Value;
++		set {
++			MaxFallSpeed = MultipliableFloat.One;
++			internalGravityBase = value;
++		}
++	}
++
++	/// <summary>
++	/// The effect of the types of wet on the NPCs gravity, here for reference when modifying gravity:
++	/// <br> wet, honetWet, and shimmerWet.</br>
++	/// <br> These are mutually exclusive and in reverse order priority where lava is the same as water. </br>
++	///	</summary>
++	public static float[] GravityWetMultipliers => new float[] { 2f / 3f, 1f / 3f, 1.5f / 3f }; //TML: #AccessibleNPCGravity, added
++
++	/// <summary>
++	/// The effect of the types of wet on the NPCs maxFallSpeed, here for reference when modifying maxFallSpeed:
++	/// <br> wet, honetWet, and shimmerWet.</br>
++	/// <br> These are mutually exclusive and in reverse order priority where lava is the same as water. </br>
++	///	</summary>
++	public static float[] MaxFallSpeedWetMultipliers => new float[] { 0.7f, 0.4f, 0.55f }; //TML: #AccessibleNPCGravity, added
++
++	/// <summary>
++	/// Set true for the NPC's gravity to be unaffected by the changes made for types 258, 425, 576, 577, 427, 426, 541, and the aiStyle 7, will have weird effects on these if used ignorantly
++	/// </summary>
++	public bool GravityIgnoresType = false; //TML: #AccessibleNPCGravity, added
++
++	/// <summary>
++	/// Set true for the NPC's gravity to be unaffected by space's lowered gravity.
++	/// </summary>
++	public bool GravityIgnoresSpace = false; //TML: #AccessibleNPCGravity, added
++
++	/// <summary>
++	/// Set true for the NPC's gravity to be unaffected by liquid's gravity.
```

Comment:
Remove the final `gravity`. Liquid doesn't really have gravity. Feel free to explain it some other way (viscosity, slowdown, etc)